packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -317,9 +324,62 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
   }
 
   a11yStatusTimeout: any
+  charKeysPressedTimeout: any
 
   componentWillUnmount() {
     clearTimeout(this.a11yStatusTimeout)
+    clearTimeout(this.charKeysPressedTimeout)
+  }
+
+  /**
+   * Used to compute the filtered items (by value and search query) and, if needed,
+   * their string equivalents, in order to be used throughout the component.
+   * @param props
+   * @param state
+   */
+  static getDerivedStateFromProps(props: DropdownProps, state: DropdownState) {
+    const { items, itemToString, multiple, search } = props
+    const { value } = state
+
+    if (!items) {
+      return state
+    }
+
+    const filteredItemsByValue = multiple
+      ? _.difference(items, value as ShorthandCollection)
+      : items
+    let filteredResults
+
+    if (search) {
+      const { itemToString } = props
+      const { searchQuery } = state
+
+      if (_.isFunction(search)) {
+        filteredResults = {
+          filteredItems: search(filteredItemsByValue, searchQuery),
+        }
```

Comment:
As we have already used early return, I suggest to use it there, too

```tsx
if (_.isFunction(search)) {
  return { filteredItems: search(filteredItemsByValue, searchQuery) }
}
```

Reply:
I thought it's better to have it assigned to that variable in branch, and return at the end `{...state, ...filteredResults }`. Now with return we need, on each branch, to return `{...state, filteredItems: whateverResultOnTheBranch}`.
Do you think it's better?